modules/aws/ec2-files.go

```diff
@@ -15,85 +16,170 @@ type RemoteFileSpecification struct {
 	RemotePathToFileFilter map[string][]string //A map of the files to fetch, where the keys are directories on the remote host and the values are filters for what files to fetch from the directory. The filters support bash-style wildcards.
 	UseSudo                bool
 	SshUser                string
-	KeyPair                *Ec2Keypair
+	SshOptions             *SshOptions
 	LocalDestinationDir    string //base path where to store downloaded artifacts locally. The final path of each resource will include the ip of the host and the name of the immediate parent folder.
 }
 
+// Specify one of KeyPair, SshAgent, or OverrideSshAgent should be specified
+// Any more (or less) than one will result in error on use
+type SshOptions struct {
+	// Username
+	UserName string
```

Comment:
There's already an `SshUser` above. I assume one or the other should be removed?

Reply:
correct, good catch

Reply:
I'm going to double check before removing also, just to make sure nothing else touches `RemoteFileSpecification` that expects it